### PR TITLE
Conditional cosmos exec

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala [hunk 1]
```diff
@@ -66,19 +68,12 @@ final class CosmosIntegrationTestServer(
...
+    val dcosUri = loadDcosUriSystemProperty()
     val propertiesMap = additionalProperties.map { testProperty =>
-      testProperty -> systemProperty(testProperty.propertyName).get
-    }
-
-    val uriKey = "uri"
-    setClientProperty("CosmosClient", uriKey, "http://localhost:7070")
-
-    propertiesMap.foreach { case (testProperty, value) =>
-      setClientProperty(testProperty.clientName, testProperty.name, value)
```
Comment (jsancio) [inline src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala:81]: This is not needed. The integration shouldn't be required to know how Cosmos was executed. If they know how Cosmos was exec then it makes it difficult to make them re-targetable.

File: src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala [hunk 2]
```diff
@@ -66,19 +68,12 @@ final class CosmosIntegrationTestServer(
...
       .getOrElse("java")
 
-    val dcosUri = systemProperty("com.mesosphere.cosmos.dcosUri").get
+    val dcosUri = loadDcosUriSystemProperty()
     val propertiesMap = additionalProperties.map { testProperty =>
-      testProperty -> systemProperty(testProperty.propertyName).get
-    }
-
-    val uriKey = "uri"
-    setClientProperty("CosmosClient", uriKey, "http://localhost:7070")
```
Comment (jsancio) [inline src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala:78]: This side effect was moved to the component that has this information.

File: src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala [hunk 3]
```diff
@@ -118,15 +113,6 @@ final class CosmosIntegrationTestServer(
...
   }
 
-  private[this] def setClientProperty(clientName: String, key: String, value: String): Unit = {
-    val property = s"com.mesosphere.cosmos.test.CosmosIntegrationTestClient.$clientName.$key"
-    val ignored = System.setProperty(property, value)
-  }
-
-  private[this] def systemProperty(key: String): Option[String] = {
-    Option(System.getProperty(key))
-  }
```
Comment (jsancio) [inline src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala:128]: These two methods were moved to the `com.mesospehre` package object.

File: src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala (hunk 1 above)
Comment (cruhland) [reply src/main/scala/com/mesosphere/cosmos/CosmosIntegrationTestServer.scala:81]: :+1:

File: src/main/scala/com/mesosphere/cosmos/TestProperty.scala [hunk 4]
```diff
@@ -1,9 +1,8 @@
 package com.mesosphere.cosmos
 
 /**
- * Specifies how to wire a system property into both the server under test and the test client.
+ * Specifies how to wire a system property for the server under test.
  *
- * @param clientName - the name of the test client that makes use of the property value.
```
Comment (cruhland) [inline src/main/scala/com/mesosphere/cosmos/TestProperty.scala]: You removed the code that uses `clientName` -- can we get rid of the field? And at that point, maybe this entire class isn't needed anymore?

File: src/main/scala/com/mesosphere/sbt/BuildPlugin.scala [hunk 5]
```diff
@@ -142,4 +160,27 @@ object BuildPlugin extends AutoPlugin {
...
+  private def saveClientCosmosProperty(value: String): Unit = {
+    val ignored = saveSystemProperty(CosmosClientPropertyName, value)
+  }
+
+  private def loadClientCosmosProperty(): Option[String] = {
+    loadSystemProperty(CosmosClientPropertyName)
+  }
+
+  private def  saveDirectCosmosProperty(value: Boolean): Unit = {
+    val ignored = saveSystemProperty(DirectConnectionPropertyName, value.toString)
```
Comment (cruhland) [inline src/main/scala/com/mesosphere/sbt/BuildPlugin.scala]: `saveSystemProperty()` returns `Unit` so the `val ignored = ...` isn't needed.

**Review by cruhland — APPROVED** [review]
LGTM, left a few minor comments.

**Review by larioj — APPROVED** [review]
LGTM

File: src/main/scala/com/mesosphere/sbt/BuildPlugin.scala [hunk 6]
```diff
@@ -104,7 +104,6 @@ object BuildPlugin extends AutoPlugin {
   /** This should be appended to (testOptions in IntegrationTest) */
   def itTestOptions(
     javaHomeValue: Option[File],
-    classpathPrefix: Seq[File],
```
Comment (jsancio) [inline src/main/scala/com/mesosphere/sbt/BuildPlugin.scala:108]: @cruhland @larioj I made a non-backwards compatible change. We don't need this anymore now that we are not overriding the default repository through a resource file.

I am going to address your comments and merge this. The cosmos and cosmos-enterprise build will be broken until the other PRs are merged. They should be green by Monday.

**Pull request merged**